examples/mamba/mamba.py

```diff
@@ -370,30 +370,34 @@ def allocate_inference_cache(self, batch_size: int, max_seqlen: int, dtype: torc
         )
         return conv_state, ssm_state
 
-    def _get_states_from_cache(self, inference_params, batch_size: int, initialize_states: bool = False):
+    def _get_states_from_cache(self, batch_size: int, initialize_states: bool = False):
         assert self.layer_idx is not None
-        if self.layer_idx not in inference_params.key_value_memory_dict:
+        
+        store = self.get_local_store()
+        
+        if self.layer_idx not in store["key_value_memory_dict"]:
             conv_state = torch.zeros(
                 batch_size,
-                self.d_model * self.expand,
+                self.d_model * self.expand // self.tp_pg.size(),
                 self.d_conv,
                 device=self.conv1d.weight.device,
                 dtype=self.conv1d.weight.dtype,
             )
             ssm_state = torch.zeros(
                 batch_size,
-                self.d_model * self.expand,
+                self.d_model * self.expand // self.tp_pg.size(),
                 self.d_state,
                 device=self.dt_proj.weight.device,
                 dtype=self.dt_proj.weight.dtype,
                 # dtype=torch.float32,
             )
-            inference_params.key_value_memory_dict[self.layer_idx] = (
+            
+            store["key_value_memory_dict"][self.layer_idx] = (
                 conv_state,
-                ssm_state,
+                ssm_state
             )
```

Comment:
You can remove the layer_idx dependence here, and revert your changes to store (store should be instance dependent)